torch/csrc/distributed/c10d/symm_mem/CUDASymmetricMemory-inl.h

```diff
@@ -115,54 +115,44 @@ __device__ __forceinline__ void wait_signal(uint32_t* addr) {
 // Pattern 0: Ensures that all writes to symm_mem buffers from previous
 // kernels across all devices are visible to the current kernel:
 //
-//   sync_remote_blocks<std::memory_order_relaxed>(...);
+//   sync_remote_blocks<false, true>(...);
 //   __syncthreads();
 //
 // Pattern 1: Ensures that all writes to symm_mem buffers from the current
 // block are visible to all remote blocks with matching blockIdx:
 //
 //   __syncthreads();
-//   sync_remote_blocks<std::memory_order_acq_rel>(...);
+//   sync_remote_blocks<true, true>(...);
 //   __syncthreads();
 //
 // Pattern 2: Ensures that symm_mem buffers read by the current kernel are safe
 // for writing by subsequent kernels across all devices.
 //
 //   __syncthreads();
-//   sync_remote_blocks<std::memory_order_relaxed>(...);
-template <std::memory_order Sem>
+//   sync_remote_blocks<true, false>(...);
```

Comment:
does the "Pattern 2" still comment applies to `sync_remote_blocks<true, false>` ? I thought <true, false> corresponds to a release.

Reply:
It's at the end of the kernel, so hasPrevMemAccess=true, hasSubsequentMemAccess=false. We make sure that our memory transactions are visible to others by putting a signal with `release`, but since we don't read anything, we don't need `acquire` here